nixos/modules/programs/openlinkhub.nix

```
@@ -0,0 +1,52 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+let
+  cfg = config.programs.openlinkhub;
+  pkg = config.programs.openlinkhub.package;
+in
+{
+  options.programs.openlinkhub = {
+    enable = lib.mkEnableOption "openlinkhub";
+    package = lib.mkPackageOption pkgs [ "delta" "openlinkhub" ] { };
+  };
+
+  config = lib.mkIf cfg.enable {
+    users.users.openlinkhub = {
+      isSystemUser = true;
+      group = "openlinkhub";
+    };
+    users.groups.openlinkhub = { };
+
+    systemd.services.OpenLinkHub = {
+      description = "Open source interface for iCUE LINK System Hub, Corsair AIOs and Hubs";
+      after = [ "sleep.target" ];
+      wantedBy = [ "multi-user.target" ];
+      preStart = ''
+        for dir in database static web; do
+          if ! [[ -d $dir ]]; then
+            cp -r ${pkg}/share/OpenLinkHub/$dir .
+            chmod -R u+w $dir
+          fi
+        done
+      '';
+      reload = "kill -s HUB $MAINPID";
+
+      serviceConfig = {
+        ExecStart = lib.getExe pkg;
```

This one probably doesn't need to be wrapped in a shellscript either

```suggestion

      serviceConfig = {
        ExecStart = lib.getExe pkg;
        ExecReload = "kill -s HUB $MAINPID";
```